### DISCUSSION TITLE
Modify config without breaking `git pull`

### DISCUSSION THREAD

**Discussion by ghost** [DominikDoom/a1111-sd-webui-tagcomplete#70] [Q&A] [answered]
Is there a good way to modify or specify a different location for the `config.json` file in the `tags` folder? Reason being that I'd like to modify it, but still be able to update this repo via the UI. I'm aware I can do something like `git stash` + `git pull` + `git stash pop`, but that defeats the purpose of using the extension manager available in the UI.

**Comment by DominikDoom** [comment]
I don't think there's a way around manually doing that at the moment. I know it's not optimal, but automatically updating without the user calling git themselves (and thus being able to stash) only became a consideration relatively recently, and I didn't have time to change it.

I guess the "proper" way now would be to only keep a defaults file on the repo and make the local config override those, using the defaults if the user hasn't added that option to his local config. But JSON isn't the greatest format for that. Still, I might give that solution a shot if I have time. 
In the meantime, the only relief I can offer is that further config updates will get increasingly unlikely, as the script already has most of the config features I could think of. And as long as the config doesn't change, there will be no conflicts with a local edited version either while updating.

The other option would be that the extension manager itself performs stashing in case of conflicts or as an opt-in setting, but that's out of my influence.

**Reply by ghost** [reply]
> I don't think there's a way around manually doing that at the moment. I know it's not optimal, but automatically updating without the user calling git themselves (and thus being able to stash) only became a consideration relatively recently, and I didn't have time to change it.

No worries! I'll try using the workaround you mentioned for now.
Maybe another option to consider is migrating the config to the settings tab in the UI, which I considered making a PR for, but I know some options like the colors wouldn't really make sense there and/or it would clutter it. I'll leave the decision to you on how to ultimately handle it.

**Reply by DominikDoom** [reply]
> Maybe another option to consider is migrating the config to the settings tab in the UI

That would definitely be an option too, my main aversion to it so far was how annoying it is to get the settings on the JavaScript side compared to parsing an external JSON file. And also that it would get pretty cluttered like you said, this script has a lot of config options compared to most. But I'll keep it in mind, maybe I can think of something.

**Comment by ghost** [comment, marked as answer]
bc80b3ea2ceb74402a60008a796f78ae37e4eb73 resolves this so I'll mark this as answered now
